src/PKSim.UI/Mappers/SimulationToSimulationReactionDiagramDTOMapper.cs

```diff
@@ -14,19 +13,24 @@ namespace PKSim.UI.Mappers
    public class SimulationToSimulationReactionDiagramDTOMapper : ISimulationToSimulationReactionDiagramDTOMapper
    {
       private readonly IReactionBuildingBlockCreator _reactionBuildingBlockCreator;
+      private readonly IDiagramModelFactory _diagramModelFactory;
 
-      public SimulationToSimulationReactionDiagramDTOMapper(IReactionBuildingBlockCreator reactionBuildingBlockCreator)
+      public SimulationToSimulationReactionDiagramDTOMapper(
+         IReactionBuildingBlockCreator reactionBuildingBlockCreator, IDiagramModelFactory diagramModelFactory)
       {
          _reactionBuildingBlockCreator = reactionBuildingBlockCreator;
+         _diagramModelFactory = diagramModelFactory;
       }
 
-      public SimulationReactionDiagramDTO MapFrom(Simulation simulation)
+      public SimulationReactionDiagramDTO MapFrom(Simulation simulation, bool recreateDiagram)
       {
-         var reactionBuildingBlockDTO = simulation.Reactions.FirstOrDefault() ?? _reactionBuildingBlockCreator.CreateFor(simulation);
+         //Note: We do not use the building block in the simulation here as it MIGHT be out of date due to 
+         //simulation being configured
+         var reactionBuildingBlock = _reactionBuildingBlockCreator.CreateFor(simulation);
```

Comment:
this was a bug here. SOmetimes the reaction building block was OFF because the configuration had changed => outdated graph

Reply:
this was another bug lol